### PR TITLE
Queue | Use VBMS ID for fetching appeal doc count

### PR DESCRIPTION
**Pull request by dannysellers**
With Caseflow Queue, we [fetch](https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/queue/QueueTable.jsx#L105) the document count per appeal directly from eFolder. In UAT and Production, though, this resulted in server errors. Ultimately, Alex [noticed](https://dsva.slack.com/archives/C54QCEHAL/p1519845913000168?thread_ts=1519844954.000427&cid=C54QCEHAL) we were passing the wrong ID when GETting.

This corrects the ID, and updates the dummy local endpoint (`/queue/docs`) to use the same header, rather than including the id in the url (`/queue/#{id}/docs`)

### AC
- [ ] In UAT and Production, Queue Table should successfully fetch document count after page load

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -118,8 +118,8 @@
     get '/tasks/:vacols_id', to: 'queue#index'
     get '/tasks/:vacols_id/*all', to: redirect('/queue/tasks/%{vacols_id}')
     get '/judges', to: 'queue#judges'
+    get '/docs', to: 'queue#document_count'
```
Comment (amprokop) [inline config/routes.rb]: Do you think we should name this endpoint such that it's obvious that it's only used in stubbed environments?

File: app/controllers/queue_controller.rb [hunk 2]
```diff
@@ -43,7 +43,8 @@ def judges
 
   def document_count
     # used for local dev. see Appeal.number_of_documents_url
-    appeal = Appeal.find(params[:appeal_id])
+    vbms_id = request.headers["HTTP_FILE_NUMBER"]
+    appeal = Appeal.find_by_vbms_id(vbms_id)
```
Comment (amprokop) [inline app/controllers/queue_controller.rb]: is `find_by_vbms_id` a method that really exists on Appeal? i don't see it

File: app/controllers/queue_controller.rb [hunk 3]
```diff
@@ -43,7 +43,8 @@ def judges
 
   def document_count
     # used for local dev. see Appeal.number_of_documents_url
-    appeal = Appeal.find(params[:appeal_id])
+    vbms_id = request.headers["HTTP_FILE_NUMBER"]
```
Comment (amprokop) [inline app/controllers/queue_controller.rb]: Think we could change the comment above to ` # only used for local dev. `, or add a `dev` to the method name? i've looked at this before and not been totally sure whether this is a production method

File: app/controllers/queue_controller.rb (hunk 2 above)
Comment (dannysellers) [reply app/controllers/queue_controller.rb]: It's a ["dynamic finder"](http://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Dynamic+attribute-based+finders), which I only know about from when I did the Rubocop upgrade a couple months ago. That doc says it's "mildly deprecated", though, whatever that means—I'll change it

**Pull request merged**